From c960885a1b0a82a94d89a967fe7d9e629b7433a3 Mon Sep 17 00:00:00 2001
From: Alok Makhariya <alok.makhariya@nxp.com>
Date: Tue, 16 Jan 2018 16:38:17 +0530
Subject: [PATCH 063/166] crypto/dpaa: retire fq while detaching with session

Signed-off-by: Alok Makhariya <alok.makhariya@nxp.com>
Acked-by: Akhil Goyal <akhil.goyal@nxp.com>
---
 drivers/crypto/dpaa_sec/dpaa_sec.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/crypto/dpaa_sec/dpaa_sec.c b/drivers/crypto/dpaa_sec/dpaa_sec.c
index fa83b84bc6f2..819aeb4e469c 100644
--- a/drivers/crypto/dpaa_sec/dpaa_sec.c
+++ b/drivers/crypto/dpaa_sec/dpaa_sec.c
@@ -1317,6 +1317,8 @@ dpaa_sec_detach_rxq(struct dpaa_sec_dev_private *qi, struct qman_fq *fq)
 
 	for (i = 0; i < qi->max_nb_sessions; i++) {
 		if (&qi->inq[i] == fq) {
+			qman_retire_fq(fq, NULL);
+			qman_oos_fq(fq);
 			qi->inq_attach[i] = 0;
 			return 0;
 		}
-- 
2.14.3

