From 614ee0af7db6fd1b035a2ea375808de8bae032db Mon Sep 17 00:00:00 2001
From: Nipun Gupta <nipun.gupta@nxp.com>
Date: Tue, 4 Jul 2017 16:19:25 +0530
Subject: [PATCH 087/166] PVT: net/dpaa2: enable TX congestion configurable

Signed-off-by: Nipun Gupta <nipun.gupta@nxp.com>
DPDK-773
---
 drivers/net/dpaa2/dpaa2_ethdev.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/net/dpaa2/dpaa2_ethdev.c b/drivers/net/dpaa2/dpaa2_ethdev.c
index fad8ae48fc0e..8051d7a36723 100644
--- a/drivers/net/dpaa2/dpaa2_ethdev.c
+++ b/drivers/net/dpaa2/dpaa2_ethdev.c
@@ -1886,6 +1886,12 @@ dpaa2_dev_init(struct rte_eth_dev *eth_dev)
 	priv->max_vlan_filters = attr.vlan_filter_entries;
 	priv->flags = 0;
 
+	/*If Tx congestion control support is not required */
+	if (getenv("DPAA2_TX_CGR_OFF")) {
+		priv->flags |= DPAA2_TX_CGR_OFF;
+		PMD_INIT_LOG(INFO, "Disable the tx congestion control support");
+	}
+
 	/* Tail drop size, td_threshold = 0 means disable it on queue */
 	if (getenv("DPAA2_RX_TAILDROP_SIZE"))
 		td_threshold = atoi(getenv("DPAA2_RX_TAILDROP_SIZE"));
-- 
2.14.3

